### PR TITLE
Add char config for LIKE wildcards.

### PR DESCRIPTION
**Pull request by dereuromark**
Resolves https://github.com/FriendsOfCake/search/issues/82

Classified as bug as currently using wildcard chars as normal text triggers wildcard behavior of DB which is not intended (unless you configure and use those specifically as wildcard chars for your filter).

2nd bug was: When using array input, the LIKE before/after was not working.
There were also no tests for this bug, this is now also fixed.

Also some small code cleanup.

### CODE REVIEW

**Comment by ADmad** [comment]
Why all the unrelated changes in a single commit?

**Comment by dereuromark** [comment]
Because I was working on a single branch for simplicity.
Those are simple fixes to easily look over and approve.

**Comment by ADmad** [comment]
You can still do separate commits for unrelated changes on a single branch :)

**Comment by dereuromark** [comment]
Or we can't be dense here and get stuff done ;)
You could just as well have fixed the issues I pointed out here yourself.

File: src/Model/Filter/Like.php [hunk 1]
```diff
@@ -15,7 +15,9 @@ class Like extends Base
         'before' => false,
         'after' => false,
         'mode' => 'or',
-        'comparison' => 'LIKE'
+        'comparison' => 'LIKE',
+        'wildcardAny' => '*',
```
Comment (ADmad) [inline src/Model/Filter/Like.php]: If default value of `wildcardAny` is kept `*` it will require a major version bump for the plugin as it's a non BC change.

File: src/Controller/Component/PrgComponent.php [hunk 2]
```diff
@@ -45,22 +45,23 @@ public function conversion($redirect = true)
             $this->request->data = $this->request->query;
             return null;
         }
-        if ($redirect) {
-            $params = $this->request->data;
+        if (!$redirect) {
+            return null;
```
Comment (jippi) [inline src/Controller/Component/PrgComponent.php]: why?

File: src/Controller/Component/PrgComponent.php [hunk 3]
```diff
@@ -77,6 +78,6 @@ protected function _actionCheck()
         if (is_string($actions)) {
             $actions = [$actions];
         }
-        return in_array($this->request->action, $actions);
+        return in_array($this->request->action, $actions, true);
```
Comment (jippi) [inline src/Controller/Component/PrgComponent.php]: why safe?

File: src/Controller/Component/PrgComponent.php (hunk 2 above)
Comment (ADmad) [inline src/Controller/Component/PrgComponent.php]: Early return, doesn't change actual behavior.

Comment (dereuromark) [inline src/Controller/Component/PrgComponent.php]: Its called return early and also called best practice :)

File: src/Controller/Component/PrgComponent.php (hunk 3 above)
Comment (dereuromark) [inline src/Controller/Component/PrgComponent.php]: why not safe?!

File: src/Model/Filter/Like.php (hunk 1 above)
Comment (dereuromark) [inline src/Model/Filter/Like.php]: True, are we fine with that? As I mentioned before no non-dev user would assume it to be % or _ anyway, they use the default wildcards from all OS and google and alike.

**Comment by dereuromark** [comment]
Can we continue?

**Comment by dereuromark** [comment]
Guys, whats the holdup?
I would very much not like to branch-off here as own composer plugin just because of that.
Let's please get things done.

**Comment by dereuromark** [comment]
Closing and forking then as split off otherwise.. :/

File: src/Model/Filter/Like.php (hunk 1 above)
Comment (josegonzalez) [inline src/Model/Filter/Like.php]: I'm fine with the bump.

File: src/Controller/Component/PrgComponent.php (hunk 3 above)
Comment (josegonzalez) [inline src/Controller/Component/PrgComponent.php]: Fine by me :)

**Comment by josegonzalez** [comment]
@dereuromark whats left on this PR, if anything?

**Comment by dereuromark** [comment]
The decision on how to make a major bump probably.
And probably how to integrate https://github.com/FriendsOfCake/search/pull/85 before a major bump for all those that would need just escaping here.

**Comment by dereuromark** [comment]
I rebased.
We can now release it with 2.0 tag and add documentation on how to use it from here on / what changed compared to 1.x.

File: src/Model/Filter/Like.php [hunk 4]
```diff
@@ -14,7 +14,8 @@ class Like extends Base
         'after' => false,
         'mode' => 'or',
         'comparison' => 'LIKE',
-        'escapeWildcards' => false
```
Comment (ADmad) [inline src/Model/Filter/Like.php]: You removed the option to disable wildcard escaping?

Comment (dereuromark) [inline src/Model/Filter/Like.php]: Of course. Not necessary anymore.

**Comment by dereuromark** [comment]
All good?

**Comment by jippi** [comment]
TL;DR LGTM

**Comment by ADmad** [comment]
TL;DR I'll make a 1.x release and then merge this to prepare for 2.0.
